src/CodeCollection/CodeChangeSet.php

```diff
@@ -24,47 +53,84 @@ class CodeChangeSet
      * Add a file change.
      *
      * @param string $path
-     * @param string $contents New contents
-     * @param string $original Original contents
+     * @param string|false $contents New contents.
+     * @param string|false $original Original contents.
+     * @param false|string $newPath New location of the file.
+     * @return void
      */
```

Comment:
Can't these params just be nullable? I like to avoid polymorphic params whenever possible, and allowing `false` implies `string|boolean`

Reply:
In this context, I need a value to indicate non-existence. (a blank file is not the same thing as non-existent file) 

I think there's value in explicitly defining that a parameter can be a string or some other non-entity.

I've switch to `null` rather `false` and I've updated the param type to say `string|null`. I've also added a bit of an explanation about what null means in this context.